### ISSUE TITLE
Duplicate identifier

### ISSUE THREAD

**Issue by DominicBoettger** [ivogabe/gulp-typescript#293]
Hello,

i want to use gulp-typescript to compile my ts files in my node projects.

I created a gulp configuration, but every time it compiles i get a lot of errors like
node_modules/mlcl_log/typings/node/node.d.ts(1419,5): error TS2300: Duplicate identifier 'export='.

I tried to solve it with 
    "noExternalResolve": true

But then the system does not find the mlcl_log module based in my node_modules directory.

I also tried to add node_modules to the sourcepath and added it manually to the reference path in my index.ts file but nothing works.

Any ideas? The same stuff works in atom with the same tsconfig.json without any problems.

Best regards
Dominic

**Comment by ivogabe** [comment]
Can you post your gulpfile and tsconfig? And can you try noResolve instead of noExternalResolve?

**Comment by DominicBoettger** [comment]
tsconfig:
```js
{
  "compileOnSave": false,
  "compilerOptions": {
    "target": "es6",
    "module": "commonjs",
    "declaration": true,
    "removeComments": true,
    "preserveConstEnums": true,
    "sourceMap": false
  },
  "filesGlob": [
    "typings/**/*.d.ts",
    "src/**/*.ts",
    "src/**/*.tsx",
    "!node_modules/**"
  ],
  "files": [
    "typings/async/async.d.ts",
    "typings/body-parser/body-parser.d.ts",
    "typings/cookie-parser/cookie-parser.d.ts",
    "typings/del/del.d.ts",
    "typings/express/express.d.ts",
    "typings/glob/glob.d.ts",
    "typings/gulp-plumber/gulp-plumber.d.ts",
    "typings/gulp-tslint/gulp-tslint.d.ts",
    "typings/gulp-typescript/gulp-typescript.d.ts",
    "typings/gulp/gulp.d.ts",
    "typings/handlebars/handlebars.d.ts",
    "typings/lodash/lodash.d.ts",
    "typings/merge2/merge2.d.ts",
    "typings/mime/mime.d.ts",
    "typings/minimatch/minimatch.d.ts",
    "typings/node/node.d.ts",
    "typings/orchestrator/orchestrator.d.ts",
    "typings/q/Q.d.ts",
    "typings/serve-static/serve-static.d.ts",
    "typings/vinyl/vinyl.d.ts",
    "src/index.ts",
    "src/lib/utils.ts"
  ],
  "atom": {
    "rewriteTsconfig": true,
    "formatOnSave": true
  },
  "formatCodeOptions": {
    "indentSize": 2,
    "tabSize": 2,
    "newLineCharacter": "\r\n",
    "convertTabsToSpaces": true,
    "insertSpaceAfterCommaDelimiter": true,
    "insertSpaceAfterSemicolonInForStatements": true,
    "insertSpaceBeforeAndAfterBinaryOperators": true,
    "insertSpaceAfterKeywordsInControlFlowStatements": true,
    "insertSpaceAfterFunctionKeywordForAnonymousFunctions": false,
    "insertSpaceAfterOpeningAndBeforeClosingNonemptyParenthesis": false,
    "placeOpenBraceOnNewLineForFunctions": false,
    "placeOpenBraceOnNewLineForControlBlocks": false
  }
}
```

gulpfile:

```ts
/// <reference path='../node_modules/gulpclass/index.d.ts'/>
import {Gulpclass, Task, SequenceTask} from 'gulpclass/Decorators';

import * as gulp from 'gulp';
import * as ts from 'gulp-typescript';
import * as plumber from 'gulp-plumber';
import * as tslint from 'gulp-tslint';
import * as del from 'del';
import * as fs from 'fs';
import * as merge from 'merge2';

@Gulpclass()
export class Gulpfile {

  config: any;
  tsProject: ts.Project;

  constructor() {
    // initial read of package data
    this.readpkgjson();
    this.tsProject = ts.createProject('tsconfig.json');
  }

  /**
   * Read package.json task
   *
   * It reads the data of the devconfig object in the package.json file
   */
  @Task('config:readpkgjson')
  readpkgjson(): any {
    var configobject = JSON.parse(fs.readFileSync('./package.json', 'utf8'));
    if (configobject && configobject.devconfig) {
      this.config = configobject.devconfig;
    }
  }

  /**
   * Clean dist directory
   */
  @Task('clean>>dist')
  clean(cb: Function) {
    return del(['./dist/**'], cb);
  }

  /**
   * Typescript lint task
   */
  @Task('ts>>lint')
  tslint() {
    let lintoptions: any = {
      emitError: false,
      sort: true,
      bell: true
    }
    return gulp.src(this.config.paths.source)
      .pipe(plumber())
      .pipe(tslint())
      .pipe(tslint.report(require('tslint-stylish'), lintoptions));
  }

  /**
   * Typescript compile task
   */
  @Task('ts>>compile')
  tscompile() {
    let sourcepaths = ['typings/**/*.d.ts'];
    sourcepaths.push(this.config.paths.source);
    var tsResult = this.tsProject.src()
      .pipe(ts(this.tsProject));

    return tsResult.js
      .pipe(gulp.dest(this.config.paths.dist));
  }

  @SequenceTask('build') // this special annotation using "run-sequence" module to run returned tasks in sequence
  build() {
    return [['clean>>dist', 'ts>>lint'], 'ts>>compile'];
  }

  @SequenceTask('watch') // this special annotation using "run-sequence" module to run returned tasks in sequence
  watch() {
    return gulp.watch(this.config.paths.source, ['build']);
  }

  @SequenceTask('default')
  default() { // because this task has "default" name it will be run as default gulp task
    return ['build', 'watch'];
  }
}
dob:mlcl_core dominic.boettger$ cat tasks/gulpclass.ts
/// <reference path='../node_modules/gulpclass/index.d.ts'/>
import {Gulpclass, Task, SequenceTask} from 'gulpclass/Decorators';

import * as gulp from 'gulp';
import * as ts from 'gulp-typescript';
import * as plumber from 'gulp-plumber';
import * as tslint from 'gulp-tslint';
import * as del from 'del';
import * as fs from 'fs';
import * as merge from 'merge2';

@Gulpclass()
export class Gulpfile {

  config: any;
  tsProject: ts.Project;

  constructor() {
    // initial read of package data
    this.readpkgjson();
    this.tsProject = ts.createProject('tsconfig.json');
  }

  /**
   * Read package.json task
   *
   * It reads the data of the devconfig object in the package.json file
   */
  @Task('config:readpkgjson')
  readpkgjson(): any {
    var configobject = JSON.parse(fs.readFileSync('./package.json', 'utf8'));
    if (configobject && configobject.devconfig) {
      this.config = configobject.devconfig;
    }
  }

  /**
   * Clean dist directory
   */
  @Task('clean>>dist')
  clean(cb: Function) {
    return del(['./dist/**'], cb);
  }

  /**
   * Typescript lint task
   */
  @Task('ts>>lint')
  tslint() {
    let lintoptions: any = {
      emitError: false,
      sort: true,
      bell: true
    }
    return gulp.src(this.config.paths.source)
      .pipe(plumber())
      .pipe(tslint())
      .pipe(tslint.report(require('tslint-stylish'), lintoptions));
  }

  /**
   * Typescript compile task
   */
  @Task('ts>>compile')
  tscompile() {
    let sourcepaths = ['typings/**/*.d.ts'];
    sourcepaths.push(this.config.paths.source);
    var tsResult = this.tsProject.src()
      .pipe(ts(this.tsProject));

    return tsResult.js
      .pipe(gulp.dest(this.config.paths.dist));
  }

  @SequenceTask('build') // this special annotation using "run-sequence" module to run returned tasks in sequence
  build() {
    return [['clean>>dist', 'ts>>lint'], 'ts>>compile'];
  }

  @SequenceTask('watch') // this special annotation using "run-sequence" module to run returned tasks in sequence
  watch() {
    return gulp.watch(this.config.paths.source, ['build']);
  }

  @SequenceTask('default')
  default() { // because this task has "default" name it will be run as default gulp task
    return ['build', 'watch'];
  }
}
```

**Comment by ivogabe** [comment]
What happens when you use `tsc` instead of gulp? Do you get the same error?

**Comment by DominicBoettger** [comment]
With tsc i get the not found error even when i don't set  "noResolve": true

**Comment by DominicBoettger** [comment]
I just published the state of my main module to a separate branch:

https://github.com/molecuel/mlcl_core/tree/gulptest

**Comment by mitchellmorris** [comment]
I am getting the same issue. I ran into subsequent and unrelated issues that lead me to update "tsd" to the new "typings" and after that it seemed that my gulp-typings start dishing out this error while `tsc -p .` runs great.

Here is my code.  Sorry if this is unrelated just throwing my 2 cents

```
gulp.task('server-ts', function () {
    g.util.log('Making SERVERSIDE JavaScript from ' + config.allServerTypeScript);
    var sourceTsFiles = [config.allServerTypeScript,
                         config.serverTypeScriptDefinitions]; //reference to library .d.ts files
                        
    var tsResult = gulp.src(sourceTsFiles)
                       .pipe(g.sourcemaps.init())
                       .pipe(g.typescript(tsServer));

    tsResult.dts.pipe(gulp.dest(config.serverTsOutDir));
    return tsResult.js
                   .pipe(g.sourcemaps.write('.'))
                   .pipe(gulp.dest(config.serverTsOutDir));
});
```

**Comment by DominicBoettger** [comment]
Alright, i solved a lot of issues by migrating to the new typings. I think my issue is now more based on structural reasons and how typings should be saved when having multiple typescript based modules in the node_modules tree. I am not sure how i could prevent from this error without getting a duplicate identifier error (when shipping my submodules with the typings e.g. node.d.ts) or when shipping without the typings i get missing typings errors for referenced filed.

Update: What i did now is to move the source file of the module to the src folder.

|_src
   |_index.ts
|_dist
   |_index.js
|_index.d.ts
|_package.json

I think this could be a solution for my problems... It doesn't scan the src folder and gets the module information from the definition file.

**Comment by DominicBoettger** [comment]
Here is my example project https://github.com/molecuel/mlcl_module_boilerplate

**Comment by ivogabe** [comment]
Sorry, I haven't had time to investigate this. I hope to do that soon with your sample project.

**Comment by ivogabe** [comment]
I tried mlcl_core, and I got the same errors with `tsc` as I got with `gulp`. Did you use the same version of TypeScript in both cases when you tested it? When I run `typings install` and then `gulp`, I get the duplicate-identifier errors. If I remove the typings directory before these commands, I get no error with gulp and no error with tsc. So it looks like that this isn't an issue with gulp-typescript. If I missed something, let me know and I'll reopen this issue.
